drivers/led/led_pwm.c

```diff
@@ -131,20 +222,23 @@ static const struct led_driver_api led_pwm_api = {
 
 #define LED_PWM_DEVICE(id)					\
 								\
-const struct led_pwm led_pwm_##id[] = {				\
+static const struct led_pwm led_pwm_##id[] = {			\
 	DT_INST_FOREACH_CHILD(id, LED_PWM)			\
 };								\
 								\
-const struct led_pwm_config led_pwm_config_##id = {		\
+static const struct led_pwm_config led_pwm_config_##id = {	\
 	.num_leds	= ARRAY_SIZE(led_pwm_##id),		\
 	.led		= led_pwm_##id,				\
 };								\
 								\
+static struct led_pwm_data					\
+	led_pwm_data_##id[ARRAY_SIZE(led_pwm_##id)];		\
```

Comment:
Why is it an array when only one such structure is used per device?

Reply:
Because of my low quality commit revert...
Original code used data for each single HW LED instance while I need only one per device. Should be fixed now. Thanks for spotting it.